File: frontend/src/services/orderService.js
import api from '../api/axios';

const orderService = {
  createOrder: async (orderData) => {
    // Backend uses @RequestParam for shippingAddress and paymentMethod
    const response = await api.post('/orders', null, { params: orderData });
    return response.data;
  },

  getUserOrders: async () => {
    const response = await api.get('/orders/my-orders');
    return response.data;
  },

  getOrderById: async (id) => {
    const response = await api.get(`/orders/${id}`);
    return response.data;
  },

  // Admin endpoints
  getAllOrders: async (params) => {
    const response = await api.get('/orders', { params });
    return response.data;
  },

  updateOrderStatus: async (id, status) => {
    const response = await api.patch(`/orders/${id}/status`, null, { params: { status } });
    return response.data;
  }
};

export default orderService;

File: src/main/java/com/ecommerce/api/service/OrderService.java
package com.ecommerce.api.service;

import com.ecommerce.api.dto.OrderDTO;
import java.util.List;

public interface OrderService {
    OrderDTO placeOrder(String userEmail, String shippingAddress, String paymentMethod);
    OrderDTO getOrderById(Long id);
    List<OrderDTO> getOrdersForCurrentUser(String userEmail);
    List<OrderDTO> getAllOrders();
    OrderDTO updateOrderStatus(Long id, String status);
}
